forge-gui/res/cardsfolder/p/perilous_predicament.txt

```diff
@@ -1,13 +1,11 @@
 Name:Perilous Predicament
 ManaCost:4 B
 Types:Instant
-A:SP$ RepeatEach | AILogic$ OpponentHasMultipleCreatures | RepeatPlayers$ Opponent | RepeatSubAbility$ DBChoose | SubAbility$ DBSacAll | SpellDescription$ Each opponent sacrifices an artifact creature and a nonartifact creature.
-SVar:DBChoose:DB$ ChooseCard | Defined$ Remembered | Amount$ 1 | Choices$ Creature.Artifact+RememberedPlayerCtrl | Mandatory$ True | RememberChosen$ True | ChoiceTitle$ Choose target artifact creature | AILogic$ WorstCard | SubAbility$ DBChoose2
-SVar:DBChoose2:DB$ ChooseCard | Defined$ Remembered | Amount$ 1 | Choices$ Creature.nonArtifact+IsNotRemembered+RememberedPlayerCtrl | Mandatory$ True | RememberChosen$ True | ChoiceTitle$ Choose target nonartifact creature | AILogic$ WorstCard
-SVar:DBSacAll:DB$ SacrificeAll | ValidCards$ Permanent.IsRemembered | SubAbility$ DBCleanup
-SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+A:SP$ RepeatEach | AILogic$ OpponentHasMultipleCreatures | RepeatPlayers$ Opponent | RepeatSubAbility$ DBSac | SpellDescription$ Each opponent sacrifices an artifact creature and a nonartifact creature.
```

Comment:
Repeat isn't needed either?

Reply:
It does work fine without `RepeatEach`. 
The only issue I have is what to do with the AILogic$ parameters of the lines to be combined. I looked into it and I take it `AILogic$ SacAndUpgrade+SacWorst` on [Natural Order](https://scryfall.com/card/ema/177/natural-order) means the relevant arguments on Perilous Predicament can be combined. However since this is the only example (on three card scripts), I'm not too sure doing so is indeed useful.

Reply:
Not having OpponentHasCreatures logic will be no real loss here
the NeedsToPlay definition is already superior

Reply:
Edited _Perilous Predicament_ and _Regna's Sanction_ accordingly.

Reply:
Oh, and I take it _Release_ of _Catch // Release_ should also be trimmed.